languages/js/src/Query.ts

```diff
@@ -113,78 +113,82 @@ export class Query {
-            callId,
-            instance,
-            args,
-          } = event.data as ExternalCall;
-          this.handleCall(attribute, callId, instance, args);
-          break;
-        }
-        case QueryEventKind.ExternalIsSubspecializer: {
-          const {
-            instanceId,
-            leftTag,
-            rightTag,
-            callId,
-          } = event.data as ExternalIsSubspecializer;
-          const answer = this.#host.isSubspecializer(
-            instanceId,
-            leftTag,
-            rightTag
-          );
-          this.questionResult(answer, callId);
-          break;
-        }
-        case QueryEventKind.ExternalIsa: {
-          const { instance, tag, callId } = event.data as ExternalIsa;
-          const answer = this.#host.isa(instance, tag);
-          this.questionResult(answer, callId);
-          break;
-        }
-        case QueryEventKind.ExternalUnify: {
-          const { leftId, rightId, callId } = event.data as ExternalUnify;
-          const answer = this.#host.unify(leftId, rightId);
-          this.questionResult(answer, callId);
-          break;
+          case QueryEventKind.ExternalCall: {
+            const {
+              attribute,
+              callId,
+              instance,
+              args,
+            } = event.data as ExternalCall;
+            this.handleCall(attribute, callId, instance, args);
+            break;
+          }
+          case QueryEventKind.ExternalIsSubspecializer: {
+            const {
+              instanceId,
+              leftTag,
+              rightTag,
+              callId,
+            } = event.data as ExternalIsSubspecializer;
+            const answer = this.#host.isSubspecializer(
+              instanceId,
+              leftTag,
+              rightTag
+            );
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.ExternalIsa: {
+            const { instance, tag, callId } = event.data as ExternalIsa;
+            const answer = this.#host.isa(instance, tag);
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.ExternalUnify: {
+            const { leftId, rightId, callId } = event.data as ExternalUnify;
+            const answer = this.#host.unify(leftId, rightId);
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.Debug:
+            const { message } = event.data as Debug;
+            if (message) console.log(message);
+            createInterface({
+              input: process.stdin,
+              output: process.stdout,
+              prompt: 'debug> ',
+              tabSize: 4,
+            }).on('line', line => {
+              const trimmed = line.trim().replace(/;+$/, '');
+              const command = this.#host.toPolar(trimmed);
+              this.#ffiQuery.debugCommand(JSON.stringify(command));
+            });
```

Comment:
Worth adding a break here in case the next person adds a new case without thinking about it?

Reply:
Done